designer/server/src/main/scala/pl/touk/nussknacker/ui/suggester/ExpressionSuggester.scala

```diff
@@ -39,11 +58,23 @@ class ExpressionSuggester(
     expression.language match {
       // currently we only support Spel and SpelTemplate expressions
       case Language.Spel | Language.SpelTemplate =>
-        spelExpressionSuggester.expressionSuggestions(
-          expression,
-          caretPosition2d.normalizedCaretPosition(expression.expression),
-          validationContext
-        )
+        spelExpressionSuggester
+          .expressionSuggestions(
+            expression,
+            caretPosition2d.normalizedCaretPosition(expression.expression),
+            validationContext
+          )
+          .flatMap {
+            case Nil =>
```

Comment:
`Nil` is not a good marker for invalid expression. We will retry suggestions even for normal cases. This heuristic should be applied as a final option, when all clean solutions can't handle it.

Reply:
This comment is still valid